tools/chplcheck/src/rules.py

```diff
@@ -116,15 +159,35 @@ def NestedCoforalls(context, node):
         return True
 
     @driver.basic_rule([Conditional, BoolLiteral, chapel.rest])
-    def BoolLitInCondStmt(context, node):
+    def BoolLitInCondStmt(context: Context, node: Conditional):
         """
         Warn for boolean literals like 'true' in a conditional statement.
         """
 
-        return False
+        lines = context.get_file_text(node.location().path()).split("\n")
+
+        cond = node.condition()
+        assert isinstance(cond, BoolLiteral)
+
+        text = None
+        if cond.value() == "true":
+            text = range_to_text(node.then_block().location(), lines)
+        elif cond.value() == "false":
+            else_block = node.else_block()
+            if else_block is not None:
+                text = range_to_text(else_block.location(), lines)
+            else:
+                text = ""
+
+        if text is not None:
+            return FullBasicRuleResult(
+                False, ChapelFixit.build(node.location(), text)
+            )
```

Comment:
My intention with `BasicRuleResult` (and the advanced versions) was that its a more complete return type. For example, it would be totally valid for a user to return `BasicRuleResult(True)` as opposed to just `True`. That said, I have removed the boolean on this because this does look a little wonky